Search review

jon-strabala opened the pull request:
I am only 1/2 done.

jon-strabala committed:
First pass on Capella review

First pass on Capella review
1) We need a better intro to the search service (I assume this doc set is for 7.2 thus the CIDR datatype)
2) you don't exclude content, you might include everything, filter mutations on a specific type, or include a JSON document tree or  a leaf child field.
3) A Search index can also analyze and modify the content in STORED in your Search index thus SUPPORTING NOVEL Search queries
4) The easiest query is a "Query String Query" and those translate easily into SQL++ in the Query/QSI workbench.
5) xref:create-search-indexes.adoc[] to xref:create-search-indexes.adoc[Create a Search Index] looks better in github (but not 100% needed).
6) "Query data using a Search Index" -or- "query data using a Search index" is a most clear and concise choice, as it directly conveys the action of querying data while emphasizing the use of a Search Index. However this will snowball into the titles and names of other documents.
7) as above xref:run-searches.adoc[Query data using a Search Index] run-searches.ado needs an update.
8) I moved the add autocomplete to a new section. We really need at least one more like: finding and highlighting text in a user reviews

jon-strabala committed:
Update search.adoc

1) need linkage and examples for "With the Couchbase SDKs"

jon-strabala committed:
Update create-search-indexes.adoc

Not I am using dev.nonprod-project-avengers.com as that is what I believe we are targeting

1) To create an index I merely
a) Supply an IndexName: mytestftsindex
b) Form the INFER picker tree on the left of the UI I right click on any collection in my case I select "hotel" under the scope "inventory" under the bucket "travel-sample"
c) Hit the " Create Index" button

2) altered xref:customize-index.adoc[] to xref:customize-index.adoc[Customize a Search Index with the Capella UI] looks better in github may not be needed
3) New section "Creating a Search Index with Couchbase SDKs" should have one example AKA: NEED_NEW_EXAMPLE_PAGE

4) The XREF page is very good (but not easy for a new dev) we should provide one example AKA: NEED_NEW_EXAMPLE_PAGE

5) altered xref:search-index-params.adoc[]  to xref:search-index-params.adoc[Search Index JSON Properties] looks better in github may not be needed

jon-strabala committed:
Update simple-search-index-payload.jsonc

"name": "travel-sample._default.gfx" is Elixir only as in names are global within a scope we call this a scoped index.

"name": "gfx" is Capella default (a "global" name) until 7.6 or 8.0 at with point things will change and Capella will default to "scoped index" but still continue to work with legacy global indexes.

Comment: Yes is is confusing but the new "scoped index" will solve a lot of problems consider tutorials that say use the name "myidx" only one person on a cluster can run the tutorial.

jon-strabala committed:
Update search-index-params.adoc

Some basic updates and clarifications

jon-strabala committed:
Update create-search-index-ui.adoc

jon-strabala committed:
Update create-search-indexes.adoc

linked to existing page

jon-strabala committed:
Update create-search-index-rest-api.adoc

clarity

jon-strabala committed:
Update simple-search-rest-api.adoc

Notes there are no "." in the specification this was a temporary ELixir thing before elixir switch to new REST endpoints for "scoped indexes"

Note
http://$CB_HOSTNAME:8094/api/index/travel-sample.inventory.landmark-content-index/query
should be 
http://$CB_HOSTNAME:8094/api/index/landmark-content-index/query

Remember right now the names in Capella are global (your index name is landmark-content-index) in the cluster no need for a bucket or a scope until 7.6 and 8.0 but this will intoduce new endpoints.

jon-strabala committed:
Update run-search-payload.sh

This is a global index not the OLD fully depreciated Elixir experiment.

jon-strabala committed:
make example actually work

jon-strabala committed:
make index name configurable

jon-strabala committed:
typo

jon-strabala committed:
Fix this it is just a template

remove UUID stuff
alter to "name": "$INDEX_NAME",

jon-strabala committed:
added example of what to query

sarahlwelton commented on modules/search/examples/autocomplete-search-index.jsonc:
```diff
@@ -1,10 +1,8 @@
+    "name": "$INDEX_NAME",
     "sourceType": "gocbcore",
     "sourceName": "$BUCKET_NAME",
-    "sourceUUID": "$BUCKET_UUID",
```
Does the sourceUUID and UUID need to be deleted for every index example? I'm confused why you deleted it for this one and not the others.

jon-strabala replied:
In short yes if you want to create an index.

If you use the REST API the UUID specifies that this is an update to an existing index - thus if you want to create an index you must delete this value.  The self managed UI requires a delete, Eben might be actually "strippng this" in his newer UI's fo ELixir and Capella V2. 

Next sourceUUID is locked to a "bucket" so if you are trying to create a new index on the same cluster it could stay in BUT if you are moving the definition to a new cluster the sourceUUID will not match the bucket name and a create index will fail.

sarahlwelton commented on modules/search/pages/create-search-indexes.adoc:
```diff
@@ -16,20 +17,33 @@ You can copy the JSON payload from the UI to create your index with the REST API
+
+Search indexes can be editied and further customized or modified to enhance search results and performance. For more details on customizing a Search Index using the Capella UI, refer to xref:customize-index.adoc[Customize a Search Index with the Capella UI].
+
+[#sdks]
```
This is a much longer-term project and I would feel better having this commented out for now.

jon-strabala replied:
It was "_After you create a Search index, you can customize it to improve search results and performance_" which wasn't quite correct.  I was just trying to correct the existing text. Not sure if it is a long term project, but it might be stuffed a bit deeper in the developer path.

sarahlwelton commented on modules/search/pages/search-query-auto-complete-code.adoc:
```diff
@@ -20,14 +20,26 @@ Create and configure a Search index that supports autocomplete, then configure y
+. xref:search-query-auto-complete-ui.adoc#search-index[Create an autocomplete Search index with the UI] or xref:create-search-index-rest-api.adoc[] with the file you created. 
 [start=2]
-. xref:simple-search-rest-api.adoc[Run a Search query] with 2-8 characters in the `query` property to test the Search index.
+. xref:simple-search-rest-api.adoc[Run a Search query] with 2-8 characters in the `query` property to test the Search index.  For the sample index "e-ngram-2-8," test the strings B, Be, Bea, Beau, and Beauf to locate the "Beaufort Hotel." Please note that the single character B is disregarded, as a minimum of two characters is required.
```
We don't know that they're using this against the travel sample index so I'm not recommending this.

jon-strabala replied:
This is such a detailed item that without a concrete example on a keyspace and a field it might fail and frustrate the user. For example if the field that is to be a autocomplete is nested three levels deep the doc text just won't work.

At a minimum the current item below should change:

13. Create a Child Field with the following settings: .....

to

13. [Optional] If needed create Child Mapping until you get to the correct level in the JSON document ....
14. Create a Child Field with the following settings: .....

sarahlwelton committed:
[AV-43732] Fixing up language to match with approved Couchbase style

jon-strabala committed:
Should be like create-search-index-ui.adoc

The correct UI for this doc set is https://dev.nonprod-project-avengers.com/sign-in
Made the first steps like create-search-index-ui.adoc
Added an optional create-child-mapping step
Tried to put the hyperlink in the menu option item (this might not work)

Opened SURF-5841 as this example is broken it is impossible to add "keyword_to_lower" as it was not added to the list in "Default Analyzer" under "General Settings"

jon-strabala committed:
Clarity: Type Identifier

jon-strabala committed:
Clarity

Needs more information to be useful

sarahlwelton commented on modules/search/pages/customize-index.adoc:
```diff
@@ -13,11 +13,13 @@ You can add the following components and configure the following options for a S
+The Type Identifier offers three optional methods for filtering documents from your collections to be included in a Search index:
 
-The type identifier tells the Search Service the document field or location in the document's ID value to use as the document's type. 
+* JSON Type Field: Selects only documents that contain a specific field with a particular string value.
```
Can we please get some clarity from say, Abhi, on what exactly the type identifier does? The equivalent parameters on the JSON, you didn't edit like this, and yet now it's apparently wrong here? I want to know what exactly this does if I've apparently gotten it wrong.

jon-strabala committed:
fix regex

sarahlwelton committed:
[AV-43732] Language cleanup again

sarahlwelton committed:
Merge branch 'capella' of https://github.com/jon-strabala/docs-devex into pr/44

jon-strabala committed:
explain when active

sarahlwelton commented on modules/search/pages/set-type-identifier.adoc:
```diff
@@ -16,19 +16,32 @@ For more information, see xref:create-search-index-ui.adoc[].
+... Under *Type Mappings*, next to the type mapping you wish to filter, go to menu:More Options (&vellip;)[Edit]:
+... add a period `.` and then the exact `string` (with no quotes) you wish to filter on to the end of the scope.collection.
+.. To use *Doc ID up to separator* to select only documents with IDs that match a prefix up to seperator string.
+... Select *Doc ID up to separator*.
```
Can we come up with a good example, here?

jon-strabala replied:
I recorded a brief video for you so you can see each of the three Type Identifier filters working visually as I explain what is going on ... the possible examples should be obvious.

*update* - redid the video - wasn't happy with the 1st pass

https://couchbase.zoom.us/rec/share/p2cCPJWJ2BPZgXn1UA9uy-IeUcE8Fwrhhq5tIUKtb6upzaoR9rmoqg-lEBSrrelg.D0S1K0hvZ7uFI8kC?startTime=1683141979000
Passcode: G0jNBL0%

_Note in the video I can't edit the Type Mapping more than one due to a bug I found today SURF-5559 which is being addressed in the Capella UI - so in that case I used the Self Manage editor hooked up to the same capella cluster._

sarahlwelton committed:
[AV-43732] Updates to set-type-identifier

sarahlwelton committed:
Merge branch 'capella' of https://github.com/jon-strabala/docs-devex into pr/44

sarahlwelton committed:
[AV-43732] Fixing heading depth

sarahlwelton committed:
[AV-43732] Clarity edits for Autocomplete topics

sarahlwelton committed:
[AV-43732] Adding examples for type identifiers to set-type-identifier.

Modifying search-index-params to include new information about type identifiers

sarahlwelton committed:
Adding content on geospatial search queries and correcting some errors

sarahlwelton committed:
Link and text cleanup + some reorganizing of auto-complete

sarahlwelton commented:
@jon-strabala , I want to get the changes in for the type identifier as this is currently published and incorrect, and has sat for long enough. Please open a new PR for any other things you want to correct with a review.
